### ISSUE TITLE
why not allow same hostname?

### ISSUE THREAD

**Issue by x94fujo6rpg** [nelsontky/gh-pages-url-shortener#96]
if anyone want to use this in your own repo (github page).  
unless you are using other domain.  
otherwise your hostname will always be `[your_github_name].github.io`  

not sure what is this check for  
```js
url.hostname === location.hostname;
```
in my case (github page)  
this check will always be true => isInvalidUrl => redirect failed  

fix:  
```js
url.hostname !== location.hostname;
```
or just delete it

**Comment by NoPlagiarism** [comment]
It used to escape redirect chains. Maybe there some way to do it better, but it is how it is now

**Comment by nelsontky** [comment]
> It used to escape redirect chains. Maybe there some way to do it better, but it is how it is now

Yup that's correct, that's the reason why! I'll take a look into it another time and leave this open!

**Comment by NullyIsHere** [comment]
Checking if the hostname includes nlsn.cf. idk
Buuuut. If its a custom domain the hostname is nlsn.cf not nelsontky.github.io.
I Think.
Bye
